test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+/* eslint no-return-assign: "off" */
+
+'use strict';
+
+const {NodeClient, WalletClient} = require('hs-client');
+const Network = require('../lib/protocol/network');
+const FullNode = require('../lib/node/fullnode');
+const MTX = require('../lib/primitives/mtx');
+const Resource = require('../lib/dns/resource');
+const rules = require('../lib/covenants/rules');
+const {types} = rules;
+
+const network = Network.get('regtest');
+const assert = require('bsert');
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const wallet = wclient.wallet('primary');
+const wallet2 = wclient.wallet('secondary');
+
+let name, cbAddress;
+const accountTwo = 'foobar';
+
+describe('Wallet HTTP', function() {
+  this.timeout(15000);
+
+  before(async () => {
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+
+    await wclient.createWallet('secondary');
+    cbAddress = (await wallet.createAddress('default')).address;
+    await wallet.createAccount(accountTwo);
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+  });
+
+  beforeEach(async () => {
+    name = await nclient.execute('grindname', [5]);
+  });
+
+  afterEach(async () => {
+    await node.mempool.reset();
+  });
+
+  it('should mine to the primary/default wallet', async () => {
+    const height = 20;
+
+    for (let i = 0; i < height; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    const info = await nclient.getInfo();
+    assert.equal(info.chain.height, height);
```

Comment:
Should we also check the generated blocks actually successfully mined to the primary/default wallet? Maybe get wallet history and check for `coinbase: true`. Same with next test. Mine as well check the height again too right?